### PR TITLE
scenegraph-node delete()

### PR DESCRIPTION
**Pull request by georgios-uber**

### CODE REVIEW

File: modules/core/src/scenegraph/nodes/scenegraph-node.js [hunk 1]
```diff
@@ -18,7 +18,11 @@ export default class ScenegraphNode {
     this._setScenegraphNodeProps(props);
   }
 
-  delete() {}
+  delete() {
```
Comment (ibgreen) [inline modules/core/src/scenegraph/nodes/scenegraph-node.js]: This should go into `Group.delete()` as that class manages `children`

By convention we also protect against multiple deletes.

```
class Group ... {
  delete() {
    this.children.forEach(child => child.delete());
    this.children = [];
    super.delete();
  }
}
```
The implementation here in the base class should be empty 

```
  delete() {}
```

The `ModelNode` needs to reimplement `delete()` to call `model.delete()`

**Comment by georgios-uber** [comment]
@ibgreen / @Pessimistress How do I delete textures and buffers?

**Comment by ibgreen** [comment]
> How do I delete textures and buffers?

@georgios-uber Call `texture.delete()` and `buffer.delete()`?

* https://github.com/uber/luma.gl/blob/master/modules/webgl/src/classes/resource.js#L60
* https://github.com/uber/luma.gl/blob/master/modules/webgl/src/classes/texture.js#L650
*https://github.com/uber/luma.gl/blob/master/modules/webgl/src/classes/buffer.js#L372

**Comment by georgios-uber** [comment]
@ibgreen / @Pessimistress Any way to verify (debug tool?) I deleted everything properly?

**Comment by ibgreen** [comment]
The stats object tracks counts of each webgl object, if you check the buffer tests you can see they verify the total number of allocated objects.

**Comment by georgios-uber** [comment]
@Pessimistress / @tsherif Vertex/Fragment shaders are not deleted when I delete the program!

**Comment by ibgreen** [comment]
> Vertex/Fragment shaders are not deleted when I delete the program!

This is because we cache and reuse compiled shaders (keyed by shader source).

We would need to implement reference counting in the `Resource` class to be able to delete those in a correct way.

**Comment by georgios-uber** [comment]
https://github.com/uber/luma.gl/pull/1017

**Comment by georgios-uber** [comment]
@ibgreen I want to rewrite this using #1017

**Comment by ibgreen** [comment]
>  I want to rewrite this using #1017

That mnakes sense, but ref counting will require a fair bit of discussion and will for sure not make it into v7, so let's land this as is for now.

**Pull request merged**